src/celeritas/global/Stepper.cc

```diff
@@ -103,6 +105,27 @@ auto Stepper<M>::operator()(SpanConstPrimary primaries) -> result_type
     return (*this)();
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Reseed the RNGs at the start of an event for "strong" reproducibility.
+ *
+ * This reinitializes the RNG states using a single seed and unique subsequence
+ * for each thread. It ensures that given an event number, that event can be
+ * reproduced.
+ */
+template<MemSpace M>
+void Stepper<M>::reseed(size_type event_id)
+{
+    if constexpr (M == MemSpace::device)
+    {
+        reseed_rng(params_->rng()->device_ref(), state_.ref().rng, event_id);
+    }
+    else
+    {
+        reseed_rng(params_->rng()->host_ref(), state_.ref().rng, event_id);
+    }
```

Comment:
From `corecel/data/Ref.hh`:
```suggestion
    reseed_rng(get_ref<M>(*params_->rng()), state_.ref().rng, event_id);
```

Reply:
thanks :)